2023/08/wtchangdm/day8.rs

```diff
@@ -0,0 +1,190 @@
+use std::collections::HashMap;
+
+#[derive(Debug, Clone, Copy)]
+enum Direction {
+    Left,
+    Right,
+}
+
+impl Direction {
+    fn from(ch: &char) -> Self {
+        match ch {
+            'L' => Self::Left,
+            'R' => Self::Right,
+            _ => panic!("come on it's impossible"),
+        }
+    }
+}
+
+struct Map {
+    cursor: usize,
+    directions: Vec<Direction>,
+    nodes: HashMap<String, [String; 2]>,
+}
+
+impl Map {
+    fn from(input: &[String]) -> Self {
```

Comment:
This is more like a constructor than a conversion. I may just call this `new` or some other thing instead.

```suggestion
    fn new(input: &[String]) -> Self {
```